src/main/java/bio/terra/service/dataset/exception/InvalidLockUsageException.java

```diff
@@ -0,0 +1,23 @@
+package bio.terra.service.dataset.exception;
+
+import bio.terra.common.exception.InternalServerErrorException;
+
+import java.util.List;
+
+public class InvalidLockUsageException extends InternalServerErrorException {
```

Comment:
nit: this seems mis-named to me. It is more of an IllegalArgument exception - the lock/unlock step was not properly constructed.

From the name, I thought it would have something to do with locking.

Reply:
Good point, will update.